specs/AppNotifications/BadgeNotificationSpec/BadgeNotifications-spec.md

```diff
@@ -0,0 +1,125 @@
+
+- As of today, Badge APIs support a limited set of glyphs. If the app doesn't find any suitable glyph to convey their status, there is no option to pin a custom overlay.
+
+More information about Windows SDK Badge API usage can be found [here](https://learn.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/badges#create-a-badge).
+
+## API proposal
+
+As part of bringing badges into Windows App SDK, we propose addressing the issues explained the previous section. With this proposal, developers will be able to pin and clear badges writing less code than what is required in the Windows SDK.
+
+### 1. Posting a numeric badge
+
+```cpp
+BadgeNotificationManager::PostValue(4);
+```
+
+Posting a badge will require a single line of code, instead of 5-6 lines that developers must write using the Windows SDK APIs.
+
+Notes:
+
+- Developers won't be able to post 0 or negative numbers for their badges.
+
+- If the specified number is greater than 99, a **99+** text will be rendered.
+
+### 2. Posting a glyph badge
+
+We will provide an enum that contains all glpyhs already supported in the Windows SDK. This enum will be named `BadgeNotificationGlyph`.
+
+```cpp
+BadgeNotificationManager::PostGlyph(BadgeNotificationGlyph::Busy);
+```
+
+For a complete list of predefined glyphs, see the IDL details section.
+
+### 3. Posting a custom overlay
+
+```cpp
+// using winrt::Windows::Foundation;
+
+std::wstring iconFilepath{ std::filesystem::current_path() / "MyCustomBadge.png" };
+BadgeNotificationManager::PostCustomBadge(Uri{iconFilepath});
+```
+
+Note: The URI is expected to be a full path.
+
+### 4. Clear a badge
+
+```cpp
+BadgeNotificationManager::Clear();
+```
+
+## Remarks
+
+- Any subsequent call from any of the `BadgeNotificationManager::Post` APIs will replace any previous badge.
+
+- Badges will be tied to the lifetime of the app window. If the app is closed, the badge will be cleared. This behavior is managed by the Taskbar.
+
+## IDL details
+
+``` idl
+namespace Microsoft.Windows.AppNotifications.Badges
+{
+    [contractversion(1)]
+    apicontract BadgeNotificationsContract {}
+
+    // The following glyphs are based on what's currently available in the Windows SDK:
+    // https://learn.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/badges
+    [contract(BadgeNotificationsContract, 1)]
+    enum BadgeNotificationGlyph
+    {
+        Activity,
+        Alarm,
+        Alert,
+        Attention,
+        Available,
+        Away,
+        Busy,
+        Error,
+        NewMessage,
+        Paused,
+        Playing,
+        Unavailable
+    };
+
+    [contract(BadgeNotificationsContract, 1)]
+    static runtimeclass BadgeNotificationManager
+    {
+        static void PostValue(UInt32 value);
```

Comment:
Why is there no Register() and Unregister() API? Do we not want to Register the App with WNS client?

Reply:
Since there is no need to talk to wpncore to do any of the WASDK badge work, I don't feel the necessity to register/unregister.

(Of course, the app should register for push or app notifications accordingly)